### PR TITLE
Update dependencies to current versions

### PR DESCRIPTION
**Pull request by DGuhr**
fix code to not use deprecated functions - `WithInsecure` is now `WithTransportCredentials(insecure.NewCredentials())`

remove old linters to avoid warnings when running `golangci-lint run --fix ` locally

make linter happy by `_`'ing out unused params

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -10,7 +10,6 @@ linters:
   enable:
     - 'bidichk'
     - 'bodyclose'
-    - 'deadcode'
```
Comment (DGuhr) [inline .golangci.yaml:13]: warnings see 
Screenshot 2023-05-09 at 12 33 53

File: middleware_test.go [hunk 2]
```diff
@@ -8,6 +8,8 @@ import (
 	"net"
 	"testing"
 
+	"google.golang.org/grpc/credentials/insecure"
+
```
Comment (vroldanbet) [inline middleware_test.go:12]: nit: modules from third-party providers should all be grouped

File: dial_opts_test.go [hunk 3]
```diff
@@ -2,6 +2,7 @@ package grpcutil_test
 
 import (
 	"log"
+	"google.golang.org/grpc/credentials/insecure"
```
Comment (vroldanbet) [inline dial_opts_test.go:5]: nit: modules from third-party providers should all be grouped

(`gofumpt` rules)

**Review by vroldanbet — COMMENTED** [review]
LGTM, left 2 nits that is going to make `gofumpt` complain

**Pull request merged**